app/config/collections/projects.php

```diff
@@ -51,6 +51,15 @@
                 'default' => null,
                 'array' => false,
             ],
+            [
+                '$id' => ID::custom('type'),
+                'type' => Database::VAR_STRING,
+                'size' => 128,
+                'required' => true,
+                'signed' => true,
+                'array' => false,
+                'filters' => [],
+            ],
```

Comment:
Let's add a migration for self-hosted for this, you can copy the latest from `Appwrite\Migration\Version` and strip out most of it, all we'll need is a `createAttribute` call passing this one in.